python/dgl/backend/tensorflow/tensor.py

```diff
@@ -0,0 +1,564 @@
+        max_len = as_scalar(lengths.max())
+    else:
+        max_len = builtins.max(lengths)
+
+    if l_min is not None:
+        max_len = builtins.max(max_len, l_min)
+
+    batch_size = len(lengths)
+    ndim = input.ndim
+    tensor_list = []
+    cum_row = 0
+    pad_nparray = np.zeros((ndim, 2), dtype=np.int32)
+    for l in lengths:
+        t = input[cum_row:cum_row+l]
+        pad_nparray[0, 1] = max_len - l
+        t = tf.pad(t, tf.constant(pad_nparray),
+                   mode='CONSTANT', constant_values=value)
+        tensor_list.append(t)
+        cum_row += l
+    return tf.stack(tensor_list, axis=0)
+
+
+def pack_padded_tensor(input, lengths):
+    out_list = []
+    for i, l in enumerate(lengths):
+        t = input[i]
+        out = t[:l]
+        out_list.append(out)
+    return tf.concat(out_list, axis=0)
+
+
+def unsorted_1d_segment_sum(input, seg_id, n_segs, dim):
+    assert dim == 0  # Why we need dim for 1d?
+    return tf.math.unsorted_segment_sum(input, seg_id, n_segs)
+
+
+def unsorted_1d_segment_mean(input, seg_id, n_segs, dim):
+    assert dim == 0  # Why we need dim for 1d?
+    return tf.math.unsorted_segment_mean(input, seg_id, n_segs)
+
+# TODO: TF has unsorted_segment_max, which can accelerate _max_on on batched graph
+
+
+def boolean_mask(input, mask):
+    return tf.boolean_mask(input, mask)
+
+
+def equal(x, y):
+    return x == y
+
+
+def logical_not(input):
+    return ~input
+
+
+def unique(input):
+    return tf.unique(input).y
+
+
+def full_1d(length, fill_value, dtype, ctx):
+    with tf.device(ctx):
+        t = tf.fill([length], value=fill_value)
+        t = tf.cast(t, dtype=dtype)
+    return t
+
+
+def nonzero_1d(input):
+    nonzero_bool = (input != False)
+    return tf.reshape(tf.where(nonzero_bool), (-1, ))
+
+
+def sort_1d(input):
+    return tf.sort(input), tf.cast(tf.argsort(input), dtype=tf.int64)
+
+
+def arange(start, stop, ctx=None):
+    if ctx is None:
+        return tf.range(start, stop, dtype=tf.int64)
+    else:
+        with tf.device(ctx):
+            t = tf.range(start, stop, dtype=tf.int64)
+        return t
+
+
+def rand_shuffle(arr):
+    return tf.random.shuffle(arr)
+
+
+def zerocopy_to_dlpack(input):
+    return tfdlpack.to_dlpack(input)
+
+
+def zerocopy_from_dlpack(dlpack_tensor):
+    # Not zero copy
+    return tfdlpack.from_dlpack(dlpack_tensor)
+
+
+def zerocopy_to_numpy(input):
+    # NOTE: not zerocopy
+    return np.array(memoryview(input))
+
+
+def zerocopy_from_numpy(np_array):
+    # NOTE: not zerocopy
+    # This assumes tensor should be on cpu
+    with tf.device("/cpu:0"):
+        t = tf.convert_to_tensor(np_array)
+    return t
+
+
+def zerocopy_to_dgl_ndarray(input):
+    return nd.from_dlpack(zerocopy_to_dlpack(input))
+
+
+def zerocopy_from_dgl_ndarray(input):
+    return zerocopy_from_dlpack(input.to_dlpack())
+
+
+def one_hot(t, num_classes=-1):
+    raise NotImplementedError
+    # return th.nn.functional.one_hot(t, num_classes)
```

Comment:
Is it used? If not, we could delete it from the backend interface.

Reply:
Sure

Reply:
Removed